[tests-only] Tus tests for uploading to shared files with checksums

swoichha opened the pull request:
## Description
This PR adds tests for:
- Uploading to shared files

## Related Issue
- https://github.com/owncloud/ocis/issues/1754
- https://github.com/owncloud/ocis/issues/1599

## How Has This Been Tested?
- :robot: 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

swoichha committed:
add tests for uploading to shared files

Talank commented on tests/acceptance/features/apiWebdavUploadTUS/uploadToShare.feature:
```diff
@@ -77,3 +77,20 @@ Feature: upload file to shared folder
+    When user "Brian" creates a new TUS resource on the WebDAV API with these headers:
+      | Tus-Resumable   | 1.0.0                                         |
+      | Upload-Length   | 16                         |
+      | Upload-Metadata | filename L1NoYXJlcy9GT0xERVIvdGV4dGZpbGUudHh0 |
```
it is hard to understand the encrypted name and would have been better if there is a comment somewhere that says what is the actual name of this file.
But, that's just a suggestion. Everything looks good to me anyway. :+1:

swoichha replied:
The encrypted file name can be seen here https://github.com/owncloud/core/pull/38484/files#diff-0613ee3f838a6348f280cfa6593798e61953aea15b0fb94b85d92335095700d9R91. 

So far for all the tests we haven't mentioned the encrypted file name(which is sent as metadata to created the file) in any comments so I don't if we should or shouldn't do it. @individual-it what do you think about this idea of mentioning encrypted filename in a comment?

individual-it replied:
I think it would be good to mention. maybe let's have an other PR doing that for all places

Talank reviewed (commented):
LGTM

individual-it reviewed (changes requested):
please also add tests where
- [x] the sharer upload a file in a shared folder and the receiver checks the checksum with propfind and in the header
- [x] upload with other checksum types
- [x] upload with chucks to a share (from both sides)
- [x] upload wrong checkssums (from both sides)

if you like some or all of those things could go in a separate PR

dpakach reviewed (commented):
apart from everything mentioned already looks good

swoichha commented:
> please also add tests where
> 
> * [x]  the sharer upload a file in a shared folder and the receiver checks the checksum with propfind and in the header
> * [x]  upload with other checksum types
> * [x]  upload with chucks to a share (from both sides)
> * [x]  upload wrong checkssums (from both sides)
> 
> if you like some or all of those things could go in a separate PR

@individual-it I have added all these tests and removed changes done in tests for `OPTIONS` request, I will add them back later on new PR as this PR claims to be about shares.

Talank commented on tests/acceptance/features/apiWebdavUploadTUS/uploadToShare.feature:
```diff
@@ -77,3 +77,153 @@ Feature: upload file to shared folder
+    And user "Alice" has created a new TUS resource on the WebDAV API with these headers:
+      | Upload-Length   | 5                                     |
+      | Upload-Metadata | filename L0ZPTERFUi90ZXh0RmlsZS50eHQ= |
+    And user "Alice" uploads file with checksum "SHA1 8cb2237d0679ca88db6464eac60da96345513954" to the last created TUS Location with offset "0" and content "uploaded content" using the TUS protocol on the WebDAV API
```
```suggestion
    When user "Alice" uploads file with checksum "SHA1 8cb2237d0679ca88db6464eac60da96345513954" to the last created TUS Location with offset "0" and content "uploaded content" using the TUS protocol on the WebDAV API
```
I think this is supposed to be a `When` step

Talank reviewed (commented):
LGTM :+1:

swoichha committed:
added extra line after each test
